proposals/0046-i18n-domain-support.md

```diff
@@ -0,0 +1,79 @@
+- Start Date: 2023/12/21
+- Reference Issues:
+- Implementation PR: https://github.com/withastro/astro/pull/9143
+
+
+# Summary
+
+First-class support for domain support in i18n routing.
+
+
+# Background & Motivation
+
+Websites that support internationalisation have different requirements. Among these requirements, there's the need to 
+have localised content under different subdomains or domains. 
+
+# Goals
+
+- Support different domains, all driven by the same Astro project;
+- Mix locales that require a different domain, with locales that don't require a different domain;
+
+# Non-Goals
+
+- Redirect users from a path website to a domain website 
+- Change website/domain based on the language of the user's browser  
```

Comment:
In this RFC, do we cover how multi-domain sites get deployed? (Goal/non-goal)

For example, the Vercel adapter could work and redirect to another subdomain, but does that mean Vercel needs to run another build for that subdomain? Or would a single build be able to share for multiple subdomains? (I don't know if Vercel supported something like this ootb)

Also, will this work for static builds? And if so, is there a specific deployment guide for it?

Reply:
No, we don't cover it, and it shouldn't be part of the proposal. Each hosting platform has its own quirks to setup multiple domains (and sub-domains). And redirects too.

The adapters **could** decide to emit the configuration needed to support subdomains, redirects, other domains, etc. although Astro doesn't need to expose anything that comes out from this proposal because it already offers everything that an adapter needs: the new configuration fields and the check for the request headers.

Reply:
> I'm not really sure if the user can easily deploy i18n-subdomain sites on Vercel yet. (or i could be wrong)

You're right! Probably the integration that will work out of the box will be the Node.js adapter